src/tests/system/tests/test_memory_cache.py

```diff
@@ -1656,3 +1659,366 @@ def test_memory_cache__truncate__nosigbus(client: Client, ldap: LDAP):
+    """
+    provider.group("group1").add()
+    client.sssd.start()
+    client.tools.getent.group("group1")
+
+    cache = "/var/lib/sss/db/cache_test.ldb"
+    timestamps = "/var/lib/sss/db/timestamps_test.ldb"
+    group_basedn = "name=group1@test,cn=groups,cn=test,cn=sysdb"
+    ldb1 = client.ldb.search(cache, group_basedn)
+    ldb2 = client.ldb.search(timestamps, group_basedn)
+
+    assert ldb1 != {}, f"ldbsearch failed to find group1@test in {cache}"
+    assert ldb2 != {}, f"ldbsearch failed to find group1@test in {timestamps}"
+    assert ldb1[group_basedn]["dataExpireTimestamp"] != ["1"], f"dataExpireTimestamp had wrong value in {cache}"
+    assert ldb2[group_basedn]["dataExpireTimestamp"] != ["1"], f"dataExpireTimestamp had wrong value in  {timestamps}"
+
+    client.sssctl.cache_expire(everything=True)
+
+    ldb1 = client.ldb.search(cache, group_basedn)
+    ldb2 = client.ldb.search(timestamps, group_basedn)
+
+    assert ldb1[group_basedn]["dataExpireTimestamp"] == ["1"], f"did not find dataExpireTimestamp in {cache}"
+    assert ldb2[group_basedn]["dataExpireTimestamp"] == ["1"], f"did not find dataExpireTimestamp in {timestamps}"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_memory_cache__check_refresh_expired_interval_40_group(client: Client, provider: GenericProvider):
+    """
+    :title: Set refresh expired interval to 40 and verify group record updates
+    :setup:
+        1. Add group
+        2. Set refresh_expired_interval to 40 and entry_cache_timeout to 160 in the sssd.conf
+        3. Start SSSD
+        4. Execute getent group group1 to fetch cache
+        5. Expire whole cache
+        6. Wait 40 seconds
+    :steps:
+        1. Get group information using ldbsearch on cache_test.ldb
+        2. Get group timestamp information using ldbsearch on timestamps_test.ldb
+    :expectedresults:
+        1. 'dataExpireTimestamp' were found in output
+        2. 'dataExpireTimestamp' had correct value in timestamps output
+    :customerscenario: False
+    """
+    provider.group("group1").add()
+    client.sssd.domain["entry_cache_timeout"] = "160"
+    client.sssd.domain["refresh_expired_interval"] = "40"
+    client.sssd.start()
+    client.tools.getent.group("group1")
+    client.sssctl.cache_expire(everything=True)
+    time.sleep(40)
+
+    cache = "/var/lib/sss/db/cache_test.ldb"
+    timestamps = "/var/lib/sss/db/timestamps_test.ldb"
+    group_basedn = "name=group1@test,cn=groups,cn=test,cn=sysdb"
+    ldb1 = client.ldb.search(cache, group_basedn)
+    ldb2 = client.ldb.search(timestamps, group_basedn)
+
+    assert ldb1[group_basedn]["dataExpireTimestamp"] == ["1"], f"did not find dataExpireTimestamp in {cache}"
+    assert ldb2[group_basedn]["dataExpireTimestamp"] != ["1"], f"dataExpireTimestamp had wrong value in {timestamps}"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_memory_cache__modify_user_invalidate_cache(client: Client, provider: GenericProvider):
```

Comment:
This can be consolidated with test_memory_cache__delete_user_updates_cache

With the topologies, 2 tests are really 10. Create one test with two users
 
user-1 - modify attribute "modify_user_invalidate_cache"
user-2 - delete "delete_user_updates_cache" 

Reducing 10 tests to 5. With the delete scenario, instead of searching the database, an authentication attempt will make it more functional.